tools/topology/topology2/cavs-nocodec-multicore.conf

```diff
@@ -0,0 +1,464 @@
+<searchdir:include>
+<searchdir:include/common>
+<searchdir:include/components>
+<searchdir:include/dais>
+<searchdir:include/pipelines/cavs>
+<searchdir:platform>
+<searchdir:platform/intel>
+
+<vendor-token.conf>
+<manifest.conf>
+<pdm_config.conf>
+<tokens.conf>
+<virtual.conf>
+<host-gateway-capture.conf>
+<io-gateway-capture.conf>
+<host-copier-gain-mixin-playback.conf>
+<mixout-gain-dai-copier-playback.conf>
+<gain-module-copier.conf>
+<gain-capture.conf>
+<gain-copier-capture.conf>
+<mixout-mixin.conf>
+<data.conf>
+<pcm.conf>
+<pcm_caps.conf>
+<fe_dai.conf>
+<ssp.conf>
+<dmic.conf>
+<intel/hw_config_cardinal_clk.conf>
+<manifest.conf>
+<route.conf>
+<intel/common_definitions.conf>
+<copier.conf>
+<module-copier.conf>
+<pipeline.conf>
+<dai.conf>
+<host.conf>
+<kpb.conf>
+<input_pin_binding.conf>
+<output_pin_binding.conf>
+<input_audio_format.conf>
+<output_audio_format.conf>
+<dmic-default.conf>
+
+Define {
+	MCLK 				24576000
+	PLATFORM 			"none"
+}
+
+# override defaults with platform-specific config
+IncludeByKey.PLATFORM {
+	"tgl"	"platform/intel/tgl.conf"
+	"adl"	"platform/intel/tgl.conf"
+	"mtl"	"platform/intel/mtl.conf"
+}
+
+#
+# List of all DAIs
+#
+Object.Dai.SSP [
+	{
+		id 		0
+		dai_index	0
+		direction	"duplex"
+		name		NoCodec-0
+		default_hw_conf_id	0
+		sample_bits		32
+		quirks			"lbm_mode"
+		io_clk		$MCLK
+
+		Object.Base.hw_config.1 {
+			name	"SSP0"
+			id	0
+			bclk_freq	3072000
+			tdm_slot_width	32
+			# TODO: remove this. Needs alsaptlg change.
+			Object.Base.link_config.1 {
+				clock_source	1
+			}
+		}
+	}
+	{
+		id 		2
+		dai_index	2
+		direction	"duplex"
+		name		NoCodec-2
+		default_hw_conf_id	0
+		sample_bits		32
+		quirks			"lbm_mode"
+		io_clk		$MCLK
+
+		Object.Base.hw_config.1 {
+			name	"SSP2"
+			id	0
+			bclk_freq	3072000
+			tdm_slot_width	32
+			# TODO: remove this. Needs alsaptlg change.
+			Object.Base.link_config.1 {
+				clock_source	1
+			}
+		}
+	}
+]
+
+#
+# Pipeline definitions
+#
+# PCM0 ---> gain ---> Mixin ---> Mixout ---> gain ---> SSP0 (core 0)
+# PCM1 ---> gain ---> Mixin ---> Mixout ---> gain ---> SSP1 (core 1)
+# PCM2 ---> gain ---> Mixin ---> Mixout ---> gain ---> SSP2 (core 2)
+# SSP0 ---> PCM0
+# SSP1 ---> PCM1
+# SSP2 ---> PCM2
+
+# Pipeline ID:1 PCM ID: 0
+Object.Pipeline.host-copier-gain-mixin-playback [
+	{
+		index	1
+
+		Object.Widget.copier.1 {
+			stream_name 'SSP0 Playback'
+		}
+		Object.Widget.gain.1 {
+			Object.Control.mixer.1 {
+				name 'Playback Volume 1'
+			}
+		}
+	}
+	{
+		index	5
+
+		core 2
```

Comment:
why would we hard-code the core values?

If you look at the topology1 nocodec topology, we set the core values with macros so that we could quickly scale down to a single core, or conversely spread all pipelines on different cores.